jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenLayerConfigurations.java

```diff
@@ -91,6 +111,73 @@ static JavaLayerConfigurations getForProject(
     return layerBuilder.build();
   }
 
+  /**
+   * Resolves the source files configuration for a War {@link MavenProject}.
+   *
+   * @param project the {@link MavenProject}
+   * @param extraDirectory path to the directory for the extra files layer
+   * @param appRoot root directory in the image where the app will be placed
+   * @return a {@link JavaLayerConfigurations} for the project
+   * @throws IOException if collecting the project files fails
+   */
+  private static JavaLayerConfigurations getForWarProject(
+      MavenProject project, Path extraDirectory, AbsoluteUnixPath appRoot) throws IOException {
+
+    Path explodedWarPath =
+        Paths.get(project.getBuild().getDirectory()).resolve(project.getBuild().getFinalName());
+    AbsoluteUnixPath dependenciesExtractionPath =
+        appRoot.resolve(JavaLayerConfigurations.WEB_INF_LIB_RELATIVE_PATH);
+    AbsoluteUnixPath classesExtractionPath =
+        appRoot.resolve(JavaLayerConfigurations.WEB_INF_CLASSES_RELATIVE_PATH);
+
+    Builder layerBuilder = JavaLayerConfigurations.builder();
+
+    // Gets all the dependencies.
+    Predicate<Path> isSnapshotDependency =
+        path -> path.toString().contains(JavaLayerConfigurations.SNAPSHOT_FILENAME_SUFFIX);
+    addFilesToLayer(
+        explodedWarPath.resolve(JavaLayerConfigurations.WEB_INF_LIB_RELATIVE_PATH),
+        isSnapshotDependency,
+        dependenciesExtractionPath,
+        layerBuilder::addSnapshotDependencyFile);
+    addFilesToLayer(
+        explodedWarPath.resolve(JavaLayerConfigurations.WEB_INF_LIB_RELATIVE_PATH),
+        isSnapshotDependency.negate(),
+        dependenciesExtractionPath,
+        layerBuilder::addDependencyFile);
+
+    // Gets the classes files in the 'WEB-INF/classes' output directory.
+    Predicate<Path> isClassFile = path -> path.toString().endsWith(CLASS_EXTENSION);
+    addFilesToLayer(
+        explodedWarPath.resolve(JavaLayerConfigurations.WEB_INF_CLASSES_RELATIVE_PATH),
+        isClassFile,
+        classesExtractionPath,
+        layerBuilder::addClassFile);
+
+    // Gets the resources
+    Predicate<Path> isResources =
+        path -> {
+          boolean inWebInfClasses =
+              path.startsWith(
+                  explodedWarPath.resolve(JavaLayerConfigurations.WEB_INF_CLASSES_RELATIVE_PATH));
+          boolean inWebInfLib =
+              path.startsWith(
+                  explodedWarPath.resolve(JavaLayerConfigurations.WEB_INF_LIB_RELATIVE_PATH));
+          boolean isClass = path.toString().endsWith(CLASS_EXTENSION);
```

Comment:
`IsClassFile` above, so no need here. Maybe we can have this predicate as a static class method and share it with `getForProject` too.

Reply:
Done